src/celeritas/optical/surface/SurfacePhysicsView.hh

```diff
@@ -219,260 +100,60 @@ CELER_FUNCTION SubsurfaceDirection SurfacePhysicsView::orientation() const
- *
- * Returns true if there's a valid surface ID, otherwise false.
- */
-CELER_FUNCTION bool SurfacePhysicsView::is_crossing_boundary() const
-{
-    return static_cast<bool>(this->surface());
-}
-
-//---------------------------------------------------------------------------//
-/*!
- * Whether the direction is exiting the surface.
- */
-CELER_FUNCTION bool SurfacePhysicsView::is_exiting(SubsurfaceDirection d) const
-{
-    // Use unsigned underflow when moving reverse (-1) on the pre-surface
-    // (position 0) to wrap to an invalid position value
-    return static_cast<size_type>(this->subsurface_position().get()
-                                  + to_signed_offset(d))
-           >= this->num_positions();
-}
-
-//---------------------------------------------------------------------------//
-/*!
- * Whether the track is in the pre-volume.
- */
-CELER_FUNCTION bool SurfacePhysicsView::in_pre_volume() const
-{
-    return this->subsurface_position().get() == 0;
-}
-
-//---------------------------------------------------------------------------//
-/*!
- * Whether the track is in the post-volume.
- */
-CELER_FUNCTION bool SurfacePhysicsView::in_post_volume() const
-{
-    return this->subsurface_position().get() + 1 == this->num_positions();
-}
-
-//---------------------------------------------------------------------------//
-/*!
- * Current position of the track in the sub-surfaces, in track-local
- * coordinates.
- *
- * Tracks traverse a surface in track-local coordinates where 0 is the
- * pre-volume and N is the post-volume. Depending on the surface orientation,
- * this will be mapped to the appropriate sub-surface material and interface.
- */
-CELER_FUNCTION SurfaceTrackPosition SurfacePhysicsView::subsurface_position() const
-{
-    return states_.surface_position[track_id_];
-}
-
-//---------------------------------------------------------------------------//
-/*!
- * Set current position of the track in the sub-surfaces, in track-local
- * coordinates.
- */
-CELER_FUNCTION void
-SurfacePhysicsView::subsurface_position(SurfaceTrackPosition pos)
-{
-    CELER_EXPECT(pos < this->num_positions());
-    states_.surface_position[track_id_] = pos;
-}
-
-//---------------------------------------------------------------------------//
-/*!
- * Get number of valid track positions in the surface.
- *
- * This is equivalent to the number of interstitial sub-surface materials, plus
- * the pre-volume and post-volumes.
- */
-CELER_FUNCTION SurfaceTrackPosition::size_type
-SurfacePhysicsView::num_positions() const
-{
-    return this->surface_record().subsurface_materials.size() + 2;
-}
-
 //---------------------------------------------------------------------------//
 /*!
  * Return the subsurface material ID of the current track position.
  */
-CELER_FUNCTION OptMatId SurfacePhysicsView::subsurface_material() const
+CELER_FUNCTION OptMatId SurfacePhysicsView::material(SurfaceTrackPosition pos) const
```

Comment:
This is an excellent example of something that needs to be a member function (since it consumes private data like the pre/post volume material). Since it depends not only on the "surface" definition but also the *state* information, because we have to cache the pre/post material, this should be part of the CrossingView.